apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameterCollection.cs

```diff
@@ -280,6 +280,7 @@ private void FillSpannerInternalTypes(MapField<string, V1.Type> typeDictionary)
         /// Clones the collection.
         /// </summary>
         /// <returns>A cloned copy of this instance.</returns>
-        public SpannerParameterCollection Clone() => new SpannerParameterCollection(_innerList);
+        public SpannerParameterCollection Clone() =>
+            new SpannerParameterCollection(_innerList.Select(p => p.Clone() as SpannerParameter));
```

Comment:
I'd suggest casting rather than using `as`, on general principle: while I'd always expect it to actually give the same result, any time it *didn't* I'd prefer a useful exception immediately instead of an ArgumentNullException or NullReferenceException later on.

Reply:
Done.